Sources/GRPCCore/RPCError.swift

```diff
@@ -42,11 +42,46 @@ public struct RPCError: Sendable, Hashable, Error {
   ///   - message: A message providing additional context about the code.
   ///   - metadata: Any metadata to attach to the error.
   ///   - cause: An underlying error which led to this error being thrown.
-  public init(code: Code, message: String, metadata: Metadata = [:], cause: (any Error)? = nil) {
-    self.code = code
-    self.message = message
-    self.metadata = metadata
-    self.cause = cause
+  ///   - flatteningCauses: Whether to flatten the `causes` as long as they're ``RPCError``s
+  ///   sharing the same ``Code-swift.struct``.
+  public init(
+    code: Code,
+    message: String,
+    metadata: Metadata = [:],
+    cause: (any Error)? = nil,
+    flatteningCauses: Bool = false
+  ) {
+    if flatteningCauses {
+      var finalMessage = message
+      var finalMetadata = metadata
+      var nextCause = cause
+      while let nextRPCErrorCause = nextCause as? RPCError {
```

Comment:
I can't really think of a reason why we wouldn't want to do this unconditionally other than flattening may potentially be expensive since we're creating new errors and dealing with strings.

If we do it unconditionally, then the loop would be unnecessary because all underlying causes would have been flattened already anyways.

And yeah good idea about the overload.

Reply:
> I can't really think of a reason why we wouldn't want to do this unconditionally other than flattening may potentially be expensive since we're creating new errors and dealing with strings.

We're already creating a new `RPCError` so that part isn't an issue. Strings a potentially expensive but that's the tradeoff for more info I guess. 

> If we do it unconditionally, then the loop would be unnecessary because all underlying causes would have been flattened already anyways.

Correct.

We can always add an opt-out later if we think it's necessary.